tlv/complex.go

```diff
@@ -0,0 +1,48 @@
+package tlv
+
+import (
+	"io"
+)
+
+// IndexStatus is an enum-like type that describes what state the indexed
+// outpoint is in. Currently only two possible values.
+type IndexStatus uint8
+
+const (
+	// OutpointOpen represents an outpoint that is open in the outpoint index.
+	OutpointOpen IndexStatus = 0
+
+	// OutpointClosed represents an outpoint that is closed in the outpoint index.
+	OutpointClosed IndexStatus = 1
+)
+
+// IndexStatusType is a tlv type that denotes IndexStatus.
+const IndexStatusType Type = 2
+
+// EIndexStatus is an encoder for channeldb.IndexStatus.
+func EIndexStatus(w io.Writer, val interface{}, buf *[8]byte) error {
```

Comment:
these encode/decode methods should be moved to `channeldb`, along with the constructor.

another option is to cast the `IndexStatusType`s to `uint8`s and just use `MakePrimitiveRecord`. might cut down on some of the additional code

Reply:
Opted for this approach - much cleaner.